source/common/upstream/subset_lb.cc

```diff
@@ -421,13 +449,28 @@ SubsetLoadBalancer::findOrCreateSubset(LbSubsetMap& subsets, const SubsetMetadat
   }
 
   idx++;
+
```

Comment:
I think if you detect `list_as_any_` and list-type `pb_value` before the `subsets.find` call (above), you can iterate over the list values and perform the remainder of the original implementation inside the loop (with the change to insert LbSubsetEntryPtr values into `entries` instead of just returning one).

I suppose that means that you'd need to factor out some code (the path that executes once for a non-list or non-list_as_any_ lb and the path that executes once per list item are the same).

I say all this because I think it's easier to understand the data structures if the list items aren't exploded into multiple entries, but I could be wrong.

Reply:
I agree that exploding the list makes this a bit hard to reason about, so I'll give your suggestion a go